cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -86,55 +83,86 @@ FlagLoop:
 }
 
-// GetStatus invokes the proper protocol check based off of whether the cluster HasTLS() to get the cluster's status
-func (cluster StaticPodCluster) GetStatus() (*clientv3.StatusResponse, error) {
-	hasTLS, err := cluster.HasTLS()
+// GetStatus gets server status
+func (c GenericClient) GetStatus() (*clientv3.StatusResponse, error) {
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   c.Endpoints,
+		DialTimeout: 5 * time.Second,
+		TLS:         c.TLSConfig,
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to determine if current etcd static pod is using TLS: %v", err)
+		return nil, err
 	}
+	defer cli.Close()
 
-	var tlsConfig *tls.Config
-	if hasTLS {
-		tlsConfig, err = NewTLSConfig(cluster.CertificatesDir)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create a TLS Config using the cluster.CertificatesDir: %v", err)
-		}
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	resp, err := cli.Status(ctx, c.Endpoints[0])
+	cancel()
+	if err != nil {
+		return nil, err
 	}
 
-	return GetClusterStatus(cluster.Endpoints, tlsConfig)
+	return resp, nil
 }
 
-// NewTLSConfig generates a tlsConfig using credentials from the default sub-paths of the certificates directory
-func NewTLSConfig(certificatesDir string) (*tls.Config, error) {
-	tlsInfo := transport.TLSInfo{
-		CertFile:      filepath.Join(certificatesDir, constants.EtcdHealthcheckClientCertName),
-		KeyFile:       filepath.Join(certificatesDir, constants.EtcdHealthcheckClientKeyName),
-		TrustedCAFile: filepath.Join(certificatesDir, constants.EtcdCACertName),
+// WaitForStatus returns a StatusResponse after an inital delay and retry attempts
+func (c GenericClient) WaitForStatus(delay time.Duration, retries int, retryInterval time.Duration) (*clientv3.StatusResponse, error) {
+	fmt.Printf("[util/etcd] Waiting %v for initial delay\n", delay)
+	time.Sleep(delay)
+	for i := 0; i < retries; i++ {
+		if i > 0 {
+			fmt.Printf("[util/etcd] Waiting %v until next retry\n", retryInterval)
+			time.Sleep(retryInterval)
+		}
+		fmt.Printf("[util/etcd] Attempting to get etcd status %d/%d\n", i+1, retries)
+		resp, err := c.GetStatus()
+		if err != nil {
+			switch err {
+			case context.DeadlineExceeded:
+				fmt.Println("[util/etcd] Attempt timed out")
+			default:
+				fmt.Printf("[util/etcd] Attempt failed with error: %v\n", err)
+			}
+			continue
+		}
+		return resp, nil
 	}
-	tlsConfig, err := tlsInfo.ClientConfig()
-	if err != nil {
-		return nil, err
+	return nil, fmt.Errorf("timeout waiting for etcd cluster status")
+}
+
+// NewClient creates a new EtcdCluster client
+func NewClient(endpoints []string, caFile string, certFile string, keyFile string) (*GenericClient, error) {
+	client := GenericClient{Endpoints: endpoints}
+
+	if caFile != "" || certFile != "" || keyFile != "" {
+		tlsInfo := transport.TLSInfo{
+			CertFile:      filepath.Join(certFile),
+			KeyFile:       filepath.Join(keyFile),
+			TrustedCAFile: filepath.Join(caFile),
```

Comment:
`filepath.Join` looks like a noop here

Reply:
Yeah, I think this was a victim of copypasta, fixing.